core/trino-main/src/main/java/io/trino/operator/DeleteAndInsertMergeProcessor.java

```diff
@@ -193,11 +198,14 @@ private void addInsertRow(PageBuilder pageBuilder, List<Block> fields, int posit
         // Add the operation column == insert
         TINYINT.writeLong(pageBuilder.getBlockBuilder(dataColumnChannels.size()), causedByUpdate ? UPDATE_INSERT_OPERATION_NUMBER : INSERT_OPERATION_NUMBER);
 
+        // Add the dummy case number, delete and insert won't use it
+        INTEGER.writeLong(pageBuilder.getBlockBuilder(dataColumnChannels.size() + 1), 0);
```

Comment:
In other places you do:
```
Update only have one update case number which default is 0
```

So let's pick different number. Maybe `-1` to make it clear it is incorrect and so it should not be used.